documentation/modules/ROOT/pages/connectors/oracle.adoc

```diff
@@ -1549,3 +1549,24 @@ Using the default `0` will mine all archive logs.
 When set to `0`, log mining history is is disabled.
 
 |===
+
+[[oracle-when-things-go-wrong]]
+== Behavior when things go wrong
+
+{prodname} is a distributed system that captures all changes in multiple upstream databases; it never misses or loses an event.
+When the system is operating normally or being managed carefully then {prodname} provides _exactly once_ delivery of every change event record.
+
+If a fault does happen then the system does not lose any events.
+However, while it is recovering from the fault, it might repeat some chang events.
+In these abnormal situations, {prodname}, like Kafka, provides _at least once_ delivery of change events.
+
+The rest of this section describes how {prodname} handles various kinds of faults and problems.
+
+[id="oracle-cannot-reference-overflow-table"]
+=== ORA-25191 - Cannot reference overflow table of an index-organized table
+
+Oracle may issue this error during the snapshot phase when encountering an index-organized table (IOT).
+An IOT table is like a traditional table but it offers a variety of performance benefits.
+First verify that the appropriate grants have been applied to the base table that the IOT is associated with.
+If the error persists, make sure the SQL that is being executed is against the appropriate table; this likely will require adjustments to the connector's table include/exclude lists.
```

Comment:
That's rather vague, how could it be the wrong table? And what's the base table of an IOT? The ORA error makes it sound the other way around, that there is an auxilliary table for an IOT.

Reply:
I can see how this is somewhat vague and I can clear that up.  But to answer your questions:

Lets say we're monitoring the `inventory` schema and there is a table called `products` that was created as an index organized table.  The table would have been created using a script like:

```
create table products ( ... ) 
  organization index
  tablespace inventory
  including <colums to be added to IOT index>
  overflow tablespace iot_overflowtbs
```

Oracle creates another object in the `inventory` schema say called `PRODUCTS_IOT`.  This is the IOT table I'm speaking about and because of how our schema include/exclude lists would work, we will attempt to process not only this `PRODUCTS` table but also `PRODUCTS_IOT` as its viewed as a table as well.  

The problem is multi-faceted because:

1. `PRODUCTS_IOT` parent table refers to an overflow tablespace for which queries on the IOT know nothing about and thus lead to ORA-25191 errors.
2. `PRODUCTS_IOT` cannot be used for certain operations, such as applying locks.  Those need to be applied to the base table which in this case is `PRODUCTS` or ORA-25191 errors occur.

So essentially these IOT tables need to be excluded from the connector's configuration for table monitoring such that we only ever interact with the actual base table itself and not the organized index or the overflow tables if my understanding of the docs is accurate.

Reply:
I've attempted to clarify this section a bit more.  Let me know if you think it still need more work.

Reply:
Ok, thanks, I understand the problem now. So the FAQ entry makes sense, but can't we automatically detect and thus ignore such overflow tables ourselves? If so, that should be the preferred approach; could you please log a JIRA issue for it then? Thx!

Reply:
Logged [DBZ-3036](https://issues.redhat.com/browse/DBZ-3036) as a follow-up as requested.